### ISSUE TITLE
false errors with `isinstance` and `Intersection`

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#762]
```py
from basedtyping import Intersection


class A: ...
class B: ...

class C(A, B): ...

AB = Intersection[A, B]

print(isinstance(C(), AB))
```
